openmc/nuclide.py

```diff
@@ -93,3 +96,293 @@ def scattering(self, scattering):
+            ax.set_ylabel('Microscopic Cross Section [b]')
+        ax.legend(loc='best')
+        ax.set_xlim(Erange)
+        if self.name is not None:
+            title = 'Microscopic Cross Section for ' + self.name
+            ax.set_title(title)
+
+        return fig
+
+    def calculate_xs(self, types, temperature=294., sab_name=None,
+                     cross_sections=None):
+        """Calculates continuous-energy cross sections of a requested type
+
+        Parameters
+        ----------
+        types : Iterable of str or Integral
+            The type of cross sections to calculate; values can either be those
+            in openmc.PLOT_TYPES or integers which correspond to reaction
+            channel (MT) numbers.
+        temperature : float, optional
+            Temperature in Kelvin to plot. If not specified, a default
+            temperature of 294K will be plotted. Note that the nearest
+            temperature in the library for each nuclide will be used as opposed
+            to using any interpolation.
+        sab_name : str, optional
+            Name of S(a,b) library to apply to MT=2 data when applicable.
+        cross_sections : str, optional
+            Location of cross_sections.xml file. Default is None.
+
+        Returns
+        -------
+        E : numpy.array
+            Energies at which cross sections are calculated, in units of eV
+        data : numpy.ndarray
+            Cross sections calculated at the energy grid described by unionE
+
+        """
+
+        # Check types
+        if cross_sections is not None:
+            cv.check_type('cross_sections', cross_sections, str)
+        if sab_name:
+            cv.check_type('sab_name', sab_name, str)
+
+        # Parse the types
+        mts = []
+        ops = []
+        yields = []
+        for line in types:
+            if line in openmc.plot_data.PLOT_TYPES:
+                mts.append(PLOT_TYPES_MT[line])
+                yields.append(PLOT_TYPES_YIELD[line])
+                ops.append(PLOT_TYPES_OP[line])
+            else:
+                # Not a built-in type, we have to parse it ourselves
+                cv.check_type('MT in types', line, Integral)
+                cv.check_greater_than('MT in types', line, 0)
+                mts.append((line,))
+                yields.append((False,))
+                ops.append(())
+
+        # If cross_sections is None, get the cross sections from the
+        # OPENMC_CROSS_SECTIONS environment variable
+        if cross_sections is None:
+            cross_sections = os.environ.get('OPENMC_CROSS_SECTIONS')
+
+        # If a cross_sections library is present, check natural nuclides
+        # against the nuclides in the library
+        if cross_sections is not None:
+            library = openmc.data.DataLibrary.from_xml(cross_sections)
+        else:
+            raise ValueError("cross_sections or OPENMC_CROSS_SECTIONS "
+                             "environmental variable must be set")
```

Comment:
Lines 249-260 are now used in several places in the python API. Could you make a general function somewhere so we can avoid code replication?